cub/cub/util_device.cuh

```diff
@@ -49,14 +49,12 @@
 
 #include <cub/detail/device_synchronize.cuh>
 #include <cub/util_debug.cuh>
-#include <cub/util_ptx.cuh>
 #include <cub/util_type.cuh>
+#include <cub/util_vsmem.cuh>
 // for backward compatibility
 #include <cub/util_temporary_storage.cuh>
 
-#include <cuda/discard_memory>
```

Comment:
@gevtushenko can you please give this refactoring a look?

Reply:
What's the story with the changes to `util_device.cuh`?

Reply:
Just including it in the right places works